### PR TITLE
add command line argument to specify which jobs will be run

### PR DESCRIPTION
**Pull request by dbro**
Hi Thomas- this is a simple change that might be useful in situations where some jobs should be run at different intervals. For example, if only jobs 2,4,and 6 should be run hourly and all the rest should run once per day, then this pair of crontab lines would do that:

05 0-2,4-23   * * * /home/pi/.local/bin/urlwatch 2 4 6
05 3   * * * /home/pi/.local/bin/urlwatch

For small sets of jobs, this could be an adequate solution for the use cases described in issues #148 #171 #331 .

### CODE REVIEW

**Comment by mborsetti** [comment]
This has been added to the `webchanges` fork: https://webchanges.readthedocs.io/en/stable/changelog.html#version-3-6-0

**Comment by thp** [comment]
The job index is kind of volatile. Wonder if we should have e.g. an optional `groups` key on each job (a list of strings) and you can then run urlwatch only for a specific group?

**Comment by dbro** [comment]
I think the syntax for this should match how the JOB is described for the commands --test-filter and --test-diff-filter . I noticed that they can match by either index numbers or (full) url values. However those commands are probably only used in developing and debugging job settings, and so they would only need to select one job at a time. The use case for specifying groups of jobs is not familiar to me, because I have only about 10 different jobs - so I don't think I should try and guess how to design a UI for it.

File: lib/urlwatch/tests/test_handler.py [hunk 1]
```diff
@@ -87,12 +87,10 @@ def test_pep8_conformance():
...
 
 
-class ConfigForTest(BaseConfig):
+class ConfigForTest(CommandConfig):
     def __init__(self, config, urls, cache, hooks, verbose):
         (prefix, bindir) = os.path.split(os.path.dirname(os.path.abspath(sys.argv[0])))
-        super().__init__('urlwatch', os.path.dirname(__file__), config, urls, cache, hooks, verbose)
-        self.edit = False
-        self.edit_hooks = False
+        super().__init__('urlwatch', os.path.dirname(__file__), bindir, prefix, config, urls, hooks, cache, verbose)
```
Comment (thp) [inline lib/urlwatch/tests/test_handler.py:93]: This change is unrelated to the PR?

File: lib/urlwatch/worker.py [hunk 2]
```diff
@@ -53,10 +53,10 @@ def run_parallel(func, items):
 def run_jobs(urlwatcher):
     cache_storage = urlwatcher.cache_storage
     jobs = [job.with_defaults(urlwatcher.config_storage.config)
-            for job in urlwatcher.jobs]
+            for (idx, job) in enumerate(urlwatcher.jobs) if ((idx + 1) in urlwatcher.urlwatch_config.joblist or (len(urlwatcher.urlwatch_config.joblist) == 0))]
```
Comment (thp) [inline lib/urlwatch/worker.py:56]: Might be good before filtering the list of jobs to filter out:

    if not all(1 <= idx <= len(urlwatcher.jobs) for idx in urlwatcher.urlwatch_config.joblist):
        raise ValueError(f'All job indices must be between 1 and {len(urlwatcher.jobs)}: {urlwatcher.urlwatch_config.joblist}')

Comment (thp) [reply lib/urlwatch/worker.py:56]: Also `len(urlwatcher.urlwatch_config.joblist) == 0` should be the same as `not urlwatcher.urlwatch_config.joblist`

**Review by thp — CHANGES_REQUESTED** [review]
See comments. I guess we could just put it in with the indexes for now. Will be hard to reshuffle this if the jobs file is edited, but that is up to the user then.

Can you add documentation for this feature too?

File: lib/urlwatch/tests/test_handler.py (hunk 1 above)
Comment (dbro) [reply lib/urlwatch/tests/test_handler.py:93]: This fixed some errors when running the tests, because with CommandConfig.__init__() the parse_args() function gets executed, but with the BaseConfig.__init__() it does not.

File: lib/urlwatch/worker.py (hunk 2 above)
Comment (dbro) [reply lib/urlwatch/worker.py:56]: Good ideas. Should I create a new commit with these suggestions?

**Comment by dbro** [comment]
Sure, done. The pull request has been updated with 2 new commits.

**Review by thp — APPROVED** [review]
Nice, thanks!
